service/src/main/java/bio/terra/pipelines/common/utils/StairwayFailedMetricsCounterHook.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.pipelines.common.utils;
+
+import static bio.terra.pipelines.common.utils.FlightUtils.inputParametersContainTrue;
+import static bio.terra.pipelines.common.utils.FlightUtils.isContextInvalid;
+
+import bio.terra.pipelines.app.common.MetricsUtils;
+import bio.terra.pipelines.dependencies.stairway.JobMapKeys;
+import bio.terra.pipelines.stairway.imputation.RunImputationJobFlightMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightStatus;
+import bio.terra.stairway.HookAction;
+import bio.terra.stairway.StairwayHook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * A {@link StairwayHook} that logs a pipeline failure to Prometheus metrics upon flight failure.
+ */
+@Component
+public class StairwayFailedMetricsCounterHook implements StairwayHook {
+  private static final Logger logger =
+      LoggerFactory.getLogger(StairwayFailedMetricsCounterHook.class);
+
+  @Override
+  public HookAction endFlight(FlightContext context) {
+    if (isContextInvalid(context)) {
+      return HookAction.CONTINUE;
+    }
+
+    if (inputParametersContainTrue(
+            context.getInputParameters(),
+            RunImputationJobFlightMapKeys.DO_INCREMENT_METRICS_FAILED_COUNTER_HOOK)
+        && context.getFlightStatus() != FlightStatus.SUCCESS) {
+      logger.info("Flight failed, incrementing failed flight counter");
```

Comment:
i think we may want to move this log into the `MetricsUtils.incrementPipelineRunFailed` call unless we wnat to put somethign specific about the flight context in there

Reply:
hmm, MetricsUtils currently doesn't have a logger at all, do you think we should add it?

Reply:
actually I meant to log the actual status (the way we do in the other hook) - ok to keep this log here in that case?

Reply:
good catch that it wasn't doing anything with the context!